camel/utils/info_channel.py

```diff
@@ -0,0 +1,102 @@
+# =========== Copyright 2023 @ CAMEL-AI.org. All Rights Reserved. ===========
+# Licensed under the Apache License, Version 2.0 (the “License”);
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an “AS IS” BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# =========== Copyright 2023 @ CAMEL-AI.org. All Rights Reserved. ===========
+import asyncio
+from typing import Any, Dict
+
+
+class Channel:
+    def __init__(self, name: str):
+        '''Channel class for communication between different tasks and agents.
+    
+        Args:
+            name (str): name of the channel
+        '''
+        self.name = name
+        self.input_queues: Dict[str, asyncio.Queue]= {}
+        self.output_queues: Dict[str, asyncio.Queue] = {}
+
+    def connect(self, channel):
+        '''Connect to another channel.
+        
+        Args:
+            chanel: another channel to connect
+        '''
+        self.input_queues[channel.name] = asyncio.Queue()
+        channel.input_queues[self.name] = asyncio.Queue()
+        self.output_queues[channel.name] = channel.input_queues[self.name]
+        channel.output_queues[self.name] = self.input_queues[channel.name]
```

Comment:
1. ensure channels are not connected multiple times?
2.  whether initialize a queue for both input and output will lead to inconsistencie?

Reply:
It's handled in ChannelManagement.register_channel, each channel will be registered only one time.